src/convenience/keyboard.ts

```diff
@@ -257,6 +354,120 @@ export class Keyboard {
         this.input_field_placeholder = value;
         return this;
     }
+    /**
+     * Creates a new keyboard that contains the transposed grid of buttons of
+     * this keyboard. This means that the resulting keyboard has the rows and
+     * columns flipped.
+     *
+     * Note that buttons can only span multiple columns, but never multiple
+     * rows. This means that if the given arrays have different lengths, some
+     * buttons might flow up in the layout. In these cases, transposing a
+     * keyboard a second time will not undo the first transposition.
+     *
+     * Here are some examples.
+     *
+     * ```
+     * original    transposed
+     * [  a  ]  ~> [  a  ]
+     *
+     *             [  a  ]
+     * [a b c]  ~> [  b  ]
+     *             [  c  ]
+     *
+     * [ a b ]     [a c e]
+     * [ c d ]  ~> [ b d ]
+     * [  e  ]
+     *
+     * [ a b ]     [a c d]
+     * [  c  ]  ~> [ b e ]
+     * [d e f]     [  f  ]
+     * ```
+     */
+    toTransposed() {
+        const original = this.keyboard;
+        const transposed = transpose(original);
+        return this.clone(transposed);
+    }
+    /**
+     * Creates a new keyboard with the same buttons but reflowed into a given
+     * number of columns (default: 4) as if the buttons were text elements.
+     * Optionally, you can specify how many buttons should be on the first
+     * column.
+     *
+     * This method is idempotent, so calling it a second time will effectively
+     * clone this keyboard without reordering the buttons.
+     *
+     * Here are some examples.
+     *
+     * ```
+     * original    reflowed
+     * [  a  ]  ~> [  a  ]    (4 columns)
+     *
+     *             [  a  ]
+     * [a b c]  ~> [  b  ]    (1 column)
+     *             [  c  ]
+     *
+     * [ a b ]     [a b c]
+     * [ c d ]  ~> [ d e ]    (3 columns)
+     * [  e  ]
+     *
+     * [ a b ]     [abcde]
+     * [  c  ]  ~> [  f  ]    (5 columns)
+     * [d e f]
+     *
+     * [a b c]     [  a  ]
+     * [d e f]  ~> [b c d]    (3 colums, 1 on first row)
+     * [g h i]     [e f g]
+     * [  j  ]     [h i j]
+     * ```
+     *
+     * @param columns Maximum number of buttons per row
+     * @param options Optional option for the first row
+     */
+    toReflowed(columns = 4, options = { first: columns }) {
```

Comment:
i'd change it to be something like `reverseFlow: boolean` instead of `first: number`. current implementation seems to be too specific